examples/create-react-app/src/pdf.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import pdfjs from 'pdfjs-dist';
+import pdfjsWorker from 'pdfjs-dist/build/pdf.worker.entry';
+
+pdfjs.GlobalWorkerOptions.workerSrc = pdfjsWorker;
+
+const PdfComponent = ({ url }) => {
+  const canvasRef = useRef(null)
+
+  useEffect(() => {
+    const fetchPdf = async () => {
+      const loadingTask = pdfjs.getDocument(url);
+
+      const pdf = await loadingTask.promise;
+
+      const firstPageNumber = 1;
+
+      const page = await pdf.getPage(firstPageNumber);
+
+      const scale = 1.5;
+      const viewport = page.getViewport({scale: scale});
+
+      // Prepare canvas using PDF page dimensions
+      const canvas = canvasRef.current;
+
+      const context = canvas.getContext('2d');
+      canvas.height = viewport.height;
+      canvas.width = viewport.width;
+
+      // Render PDF page into canvas context
+      const renderContext = {
+        canvasContext: context,
+        viewport: viewport
+      };
+      const renderTask = page.render(renderContext);
+
+      await renderTask.promise;
+    };
+
+    fetchPdf();
+  }, [url]);
+
+  return (
+    <canvas
+      ref={canvasRef}
+      width={window.innerWidth}
+      height={window.innerHeight}
+    />
+  );
+}
+
+PdfComponent.propTypes = {
+  url: PropTypes.string
+};
+
+PdfComponent.defaultProps = {
+  url: 'https://raw.githubusercontent.com/mozilla/pdf.js/ba2edeae/examples/learning/helloworld.pdf'
```

Comment:
Can we use `https://github.com/mozilla/pdf.js/blob/master/examples/learning/helloworld.pdf` here. Alternatively, perhaps better, would be to copy the file into the example directory and use the local file instead, to make the example not depend on GitHub at all.

Reply:
Done :)